### PR TITLE
Eks provider nested stack

### PR DESCRIPTION
**Pull request by cmcknight-vr-gs**

### CODE REVIEW

File: packages/@aws-cdk/aws-eks/lib/gs-extension/kubectl-nested-stack.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    const parentScope = new CoreConstruct(scope, id + '.NestedStack');
+
+    this.resource = new CfnStack(parentScope, `${id}.NestedStackResource`, {
+      templateUrl: props.templateURL,
+      parameters: {
+        ClusterArn: props.cluster.clusterArn,
+        ClusterCreationRoleArn: props.clusterCreationRole.roleArn,
+        SubnetIds: Fn.join(',', props.cluster.kubectlPrivateSubnets.map(subnet => subnet.subnetId)),
+        SecurityGroupIds: Fn.join(',', [props.cluster.clusterHandlerSecurityGroup.securityGroupId]),
```
Comment (cmcknight-vr-gs) [inline packages/@aws-cdk/aws-eks/lib/gs-extension/kubectl-nested-stack.ts]: Not 100% sure on `clusterHandlerSecurityGroup` here

Comment (hmmuthus-gs) [reply packages/@aws-cdk/aws-eks/lib/gs-extension/kubectl-nested-stack.ts]: Thanks @cmcknight-vr . So the Security Group is attached to all the Lambda functions

File: packages/@aws-cdk/aws-eks/lib/gs-extension/cluster-resource-nested-stack.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+export interface ClusterResourceNestedStackProps {
+  templateURL: string;
+  clusterCreationRole: iam.IRole;
+  subnets?: ec2.ISubnet[];
+  securityGroup?: ec2.ISecurityGroup;
+
+  removalPolicy?: RemovalPolicy;
+}
+
+const PROVIDER_ARN_OUTPUT_NAME = 'Outputs.ClusterResourceProviderframeworkonEventC6B02E13Arn';
```
Comment (hmmuthus-gs) [inline packages/@aws-cdk/aws-eks/lib/gs-extension/cluster-resource-nested-stack.ts:15]: Just leaving a comment to ensure the resource output of Lambda matches the ARN of the stack

**Review by hmmuthus-gs — APPROVED** [review]
Looks good and tested locally. Just to note the value of the nested stack resources ARNs should match up the template value that its referring on the templateURL

**Review by hmmuthus-gs — APPROVED** [review]

File: packages/@aws-cdk/aws-eks/lib/kubectl-handler/apply/__init__.py [hunk 3]
```diff
@@ -21,20 +21,54 @@ def apply_handler(event, context):
...
+- context:
+    cluster: {cluster_name}
+    user: lambda
+  name: default
+current-context: default
+users:
+- name: lambda
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1alpha1
```
Comment (hmmuthus-gs) [inline packages/@aws-cdk/aws-eks/lib/kubectl-handler/apply/__init__.py:60]: looks good to use iam authenticator

File: packages/@aws-cdk/aws-eks/lib/gs-extension/kubernetes-external-secrets.ts [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      });
+
+    const values = props.values ?? {};
+    values.serviceAccount = {
+      create: false,
+      name: this.serviceAccount.serviceAccountName,
+    };
+    values.env = values.env ?? {};
+    values.env.AWS_REGION = region;
+
```
Comment (hmmuthus-gs) [inline packages/@aws-cdk/aws-eks/lib/gs-extension/kubernetes-external-secrets.ts:106]: Note: the Region is needed for vpc endpoints. Also the namespace and version can be overridden as newer versions release

**Review by hmmuthus-gs — APPROVED** [review]
Great work on this

**Pull request closed without merging**